File: Python/1417.py
class Solution(object):
    def reformat(self, s):
        word = ""
        letter = [x for x in s if x.isalpha()]
        number = [x for x in s if x.isdigit()]
        if abs(len(letter) - len(number)) > 1:
            return ""
        if len(letter) > len(number):
            for i in range(len(number)):
                word += letter[i] + number[i]
            word += letter[-1]
        else:
            for i in range(len(letter)):
                word += number[i] + letter[i]
            if len(number) != len(letter):
                word += number[-1]
        return word

File: Java/1417.java
import java.util.*;
class Solution {
    public String reformat(String s) {
        List<Character> letter = new ArrayList<>();
        List<Character> number = new ArrayList<>();
        String word = "";
        for(int i = 0; i < s.length(); i ++){
            if(s.charAt(i) >= '0' && s.charAt(i) <= '9') number.add(s.charAt(i));
            else letter.add(s.charAt(i));
        }
        if(letter.size() - number.size() > 1 || number.size() - letter.size() > 1) return "";
        if(letter.size() > number.size()){
            for(int i = 0; i < number.size(); i++) word += "" + letter.get(i) + number.get(i);
            word += letter.get(letter.size()-1);
        }
        else{
            for(int i = 0; i < letter.size(); i++) word += "" + number.get(i) + letter.get(i);
            if(letter.size() != number.size()) word += number.get(number.size()-1);
        }
        return word;
    }
}